### PR TITLE
Add SimpleITK recipe to build SimpleITK for Python

### PR DESCRIPTION
**Pull request by blowekamp**
Checklist

- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/master/recipes/example/meta.yaml#L57-L66) for an example)
- [x] Source is from official source
- [x] Package does not vendor other packages
- [x] Build number is 0
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there

### CODE REVIEW

**Comment by blowekamp** [comment]
@hmaarrfk
@mariusvniekerk
@conda-forge/help-python

**Comment by blowekamp** [comment]
Odd error on Azure Pipelines when trying to run a compiled executable during the build process: "The application has failed to start because its side-by-side configuration is incorrect." I believe that is with the VS2008 for python compiler.

**Comment by blowekamp** [comment]
@zivy I have added you as a maintainer is that OK?

File: recipes/simpleitk/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+    - zlib
+    - hdf5                      # [not win]
+
+test:
+  imports:
+    - SimpleITK
+
+about:
+  home: http::/www.simpleitk.org
+  license: Apache 2.0
```
Comment (mariusvniekerk) [inline recipes/simpleitk/meta.yaml]:
```suggestion
  license: Apache 2.0
  license_file: LICENSE
```

Comment (blowekamp) [reply recipes/simpleitk/meta.yaml]: Good suggestion! Done locally.

**Comment by zivy** [comment]
> @zivy I have added you as a maintainer is that OK?

Thanks for adding me.

**Comment by blowekamp** [comment]
🎉 🎆 🍾 ☝️ 🎈 🍰 

Looks like we are still timing out for Py27 on appveyor. I disable py27 on windows then this PR will finally be done.

File: recipes/simpleitk/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+
+build:
+    number: 0
+    skip: true  # [ win and py27 ]
+
+requirements:
+  build:
+    - cmake   >=3.11
+    - ninja
+    - swig    >=3
```
Comment (scopatz) [inline recipes/simpleitk/meta.yaml]: probably should remove swig from here

Comment (blowekamp) [reply recipes/simpleitk/meta.yaml]: Without it being in both build and host, it was not able to locate the swig executable. Do you have a suggestion for setting SWIG_EXECUTABLE in the bat and sh files what would work with only a host requirement?

**Comment by blowekamp** [comment]
This recipe appear to be successfully building and packaging on azure. I think it is ready to be merged.

**Comment by isuruf** [comment]
Can you mention the third part libraries vendored here? Are their licenses packaged?

**Comment by blowekamp** [comment]
The Insight Toolkit (ITK) is included in this build/package because the Conda-forge package has been broken due to deep can complicate CMake/Anaconda compilation and rpath issues. Additionally, ITK is a highly configurable library that is usually configured for a particular projects purpose. ITK itself is also licensed under Apache 2.0. Additionally it provide the following [NOTICE](https://github.com/InsightSoftwareConsortium/ITK/blob/master/NOTICE) file. I think include that file will incorporate all third party licenses and notices.

**Comment by isuruf** [comment]
> because the Conda-forge package has been broken due to deep can complicate CMake/Anaconda compilation and rpath issues.

Do you have an issue?

If ITK is to be vendored here, you need to include the NOTICE file in the `license_file` field. (`license_file` can optionally be a YAML list)

**Comment by blowekamp** [comment]
> Is there an issue somewhere?

Here are two similar issues:
https://github.com/conda-forge/libitk-feedstock/issues/24
https://github.com/conda-forge/libitk-feedstock/issues/7

> 
> If ITK is to be vendored here, you need to include the NOTICE file in the `license_file` field. (`license_file` can optionally be a YAML list)

Will do.

**Comment by blowekamp** [comment]
@isuruf We are now trying to re-use libitk. We are getting the following error on windows:
```
ninja: error: 'C:/bld/libitk_1566344915319/_h_env/Library/lib/z.lib', needed by 'Wrapping/Python/_SimpleITK.pyd', missing and no known rule to make it
```
The libitk depends on zlib package. Inspecting the libitk package shows a hard coded path in the ITK configuration files:
```
ITKTargets.cmake:  INTERFACE_LINK_LIBRARIES "C:/bld/libitk_1566344915319/_h_env/Library/lib/z.lib"
```

Any ideas?

File: recipes/simpleitk/meta.yaml (hunk 2 above)
Comment (blowekamp) [reply recipes/simpleitk/meta.yaml]: swig is now only in the host, and BUILD_PREFIX is used to search for swig.

**Comment by blowekamp** [comment]
@conda-forge/staged-recipes This recipe is ready to go!

It has been refactored to use `libitk` and the newly created `libsimpleitk` packages.

File: recipes/simpleitk/build.sh [hunk 3]
```diff
@@ -0,0 +1,36 @@
+#!/bin/bash
```
Comment (scopatz) [inline recipes/simpleitk/build.sh]: Can you add `set -e` to this please?

**Comment by scopatz** [comment]
Looks great to me overall! Just one build suggestion

File: recipes/simpleitk/build.sh (hunk 3 above)
Comment (isuruf) [reply recipes/simpleitk/build.sh]: That is unnecessary as conda-build sets it.

**Pull request merged**